### PR TITLE
Assert a prompt before typing commands

### PR DESCRIPTION
**Pull request by Soulofdestiny**
waiting for verification run

### CODE REVIEW

**Comment by okurz** [comment]
Just to check our assumptions we should verify this. I thought you triggered a job already as well?

File: tests/console/git.pm [hunk 1]
```diff
@@ -42,6 +42,7 @@ sub run {
     type_string("yes\n");
     assert_screen("password-prompt");
     type_string("$password\n");
+    assert_screen 'root-console';
```
Comment (asmorodskyi) [inline tests/console/git.pm]: why you not using here testapi function type_password here ?

**Comment by asmorodskyi** [comment]
not sure that it's really suits here but at least it worth to take a look at exec_and_insert_password function.

**Comment by Soulofdestiny** [comment]
@okurz triggered verification job
@asmorodskyi this is not part of fixing this test on the fastlane without ticket, I'll create one for revisiting this test to improve it

**Comment by Soulofdestiny** [comment]
finally, here is a verification run: http://opeth/tests/408#step/git/1

File: tests/console/git.pm (hunk 1 above)
Comment (okurz) [reply tests/console/git.pm]: Well, should not change the behaviour we want to fix here but true, we should use `type_password`. @Soulofdestiny while you are at the module, could you please simplify it by removing line L21 and replacing `type_string(password)` with `type_password`? Could be a second commit in the same PR.

Comment (okurz) [reply tests/console/git.pm]: but it would need to be

```perl
type_password;
send_key 'ret';
```

**Comment by okurz** [comment]
I crosschecked all references to `type_password` where used in a console tests and found different ways how it's used.
There is a method called `handle_password_prompt` in *lib/susedistribution* which we could use here to replace the assert_screen before the typing and the typing itself - but not the assert_screen afterwards to check for the root terminal. There is `check_prompt_for_boot` in *lib/proxymode.pm* checking for the needle "text-logged-in-root" afterwards (instead of "root-console"). This is also used in *tests/virt_autotest/reboot_and_wait_up.pm*. So I wonder if the needles with the tags "root-console" and "text-logged-in-root" are used for the same purpose. I think we should resolve that question and then make sure we are using a common way or explicit differences. I am checking current needle uses …

File: tests/console/git.pm (hunk 1 above)
Comment (Soulofdestiny) [reply tests/console/git.pm]: changed - retriggered verification run

Comment (okurz) [reply tests/console/git.pm]: make sure to update the code then as well

**Comment by Soulofdestiny** [comment]
added commit for type_password
new verification run: http://opeth/tests/411#step/git/1

**Pull request merged**